src/backend/dx11/src/lib.rs

```diff
@@ -370,35 +423,59 @@ impl core::Device for Device {
-            |_, v| unsafe { (*v.0).Release(); }, //sampler
-            |_, _fence| {},
-            |_, _mapping| {},
-        );
+                                                   |_, buffer| unsafe {
+                                                       (*(buffer.resource().0).0).Release();
+                                                   },
+                                                   |_, s| unsafe {
+                                                       // shader
+                                                       (*s.object).Release();
+                                                       (*s.reflection).Release();
+                                                   },
+                                                   |_, program| unsafe {
+            let p = program.resource();
+            if p.vs != ptr::null_mut() {
+                (*p.vs).Release();
+            }
+            if p.hs != ptr::null_mut() {
+                (*p.hs).Release();
+            }
+            if p.ds != ptr::null_mut() {
+                (*p.ds).Release();
+            }
+            if p.gs != ptr::null_mut() {
+                (*p.gs).Release();
+            }
+            if p.ps != ptr::null_mut() {
+                (*p.ps).Release();
+            }
+        },
+                                                   |_, v| unsafe {
+            // PSO
+            type Child = *mut winapi::ID3D11DeviceChild;
+            (*v.layout).Release();
+            (*(v.rasterizer as Child)).Release();
+            (*(v.depth_stencil as Child)).Release();
+            (*(v.blend as Child)).Release();
+        },
+                                                   |_, texture| unsafe {
+                                                       (*texture.resource().to_resource())
+                                                           .Release();
+                                                   },
+                                                   |_, v| unsafe {
+                                                       (*v.0).Release();
+                                                   }, // SRV
+                                                   |_, _| {}, // UAV
+                                                   |_, v| unsafe {
+                                                       (*v.0).Release();
+                                                   }, // RTV
+                                                   |_, v| unsafe {
+                                                       (*v.0).Release();
+                                                   }, // DSV
+                                                   |_, v| unsafe {
+                                                       (*v.0).Release();
+                                                   }, // sampler
+                                                   |_, _fence| {},
+                                                   |_, _mapping| {});
```

Comment:
Something holds the mutable access to the mapping, so the `cleanup()` cannot obtain it along.